app/src/main/java/com/github/se/wanderpals/service/SessionManager.kt

```diff
@@ -35,17 +48,31 @@ object SessionManager {
    * @param role The user's role within the application, optional.
    * @param geoCords The user's geographic coordinates, optional.
    */
+  @Suppress("UselessCallOnNotNull")
   fun setUserSession(
       userId: String = currentUser?.userId ?: "",
       name: String = currentUser?.name ?: "",
       email: String = currentUser?.email ?: "",
       role: Role = currentUser?.role ?: Role.VIEWER,
       geoCords: GeoCords = currentUser?.geoCords ?: GeoCords(0.0, 0.0),
-      profilePhoto: String = currentUser?.profilePhoto ?: "",
+      profilePhoto: String = currentUser?.profilePhoto ?: default_profile_photo,
       tripName: String = currentUser?.tripName ?: "",
       nickname: String = currentUser?.nickname ?: ""
   ) {
-    currentUser = SessionUser(userId, name, email, role, geoCords, profilePhoto, tripName, nickname)
+    val profilePhotoUse = profilePhoto.isNullOrEmpty().let { default_profile_photo }
```

Comment:
val profilePhotoUse = if (profilePhoto.isNullOrEmpty()) default_profile_photo else profilePhoto
This should at least partially fix the issue, of the picture not displaying. However even with this, it is still inconsistent, and breaks.

Reply:
This unfortunately didn't work, because profilePhoto from firebase isn't actually null but "null"